### PR TITLE
[CORE] Clean up RowToColumnar & ColumnarToRow

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

This pr cleanup the RowToColumnar and ColumnarToRow. The main change is that, pull out methods: 
- output
- outputPartitioning
- outputOrdering
- doExecuteBroadcast
- doExecute
- supportsColumnar

## How was this patch tested?

Pass CI

### CODE REVIEW

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxValidator.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.execution.SparkPlan
 import org.apache.spark.sql.types.{ArrayType, BinaryType, BooleanType, ByteType, DataType, DateType, DecimalType, DoubleType, FloatType, IntegerType, LongType, MapType, ShortType, StringType, StructType, TimestampType}
 
-class Validator extends ValidatorApi {
+class VeloxValidator extends ValidatorApi {
```
Comment (ulysses-you) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxValidator.scala:28]: align with the file name

File: backends-velox/src/main/scala/io/glutenproject/execution/RowToVeloxColumnarExec.scala [hunk 2]
```diff
@@ -40,11 +38,15 @@ import org.apache.arrow.memory.ArrowBuf
...
-  extends RowToColumnarExecBase(child = child)
-  with UnaryExecNode {
+case class RowToVeloxColumnarExec(child: SparkPlan) extends RowToColumnarExecBase(child = child) {
 
   override def doExecuteColumnarInternal(): RDD[ColumnarBatch] = {
+    if (!new VeloxValidator().doSchemaValidate(schema)) {
+      throw new UnsupportedOperationException(
+        s"Input schema contains unsupported type when convert row to columnar, " +
+          s"${schema.toString()}")
+    }
```
Comment (ulysses-you) [inline backends-velox/src/main/scala/io/glutenproject/execution/RowToVeloxColumnarExec.scala:48]: move to the beginning of method

**Comment by ulysses-you** [comment]
cc @PHILO-HE @jinchengchenghh thank you

File: backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarToRowExec.scala [hunk 3]
```diff
@@ -38,7 +37,7 @@ case class VeloxColumnarToRowExec(child: SparkPlan) extends ColumnarToRowExecBas
 
   override def nodeName: String = "VeloxColumnarToRowExec"
 
-  override def buildCheck(): Unit = {
```
[the remark this answers is not in the document]
Comment (ulysses-you) [reply backends-velox/src/main/scala/org/apache/spark/sql/execution/VeloxColumnarToRowExec.scala:41]: yeah, we have removed it in this pr

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala [hunk 4]
```diff
@@ -37,7 +37,7 @@ class VeloxBackend extends Backend {
   override def iteratorApi(): IteratorApi = new IteratorHandler
   override def sparkPlanExecApi(): SparkPlanExecApi = new SparkPlanExecHandler
   override def transformerApi(): TransformerApi = new TransformerHandler
-  override def validatorApi(): ValidatorApi = new Validator
+  override def validatorApi(): ValidatorApi = new VeloxValidator
```
Comment (zhztheplayer) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:40]: Do we have a consistent naming convention for the components? Seems to me that we mixed the use of `...Handler`, `...`, `...Initializer`, `Velox...`

Comment (PHILO-HE) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:40]: Binwei once asked us to remove "velox" prefix inside velox module (maybe, only few exceptions not), for brevity consideration. cc @FelixYBW

Comment (ulysses-you) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala]: This change is due to we have a file `VeloxValidator` but with a different class name `Validator`. This pr changes the class name to `VeloxValidator`. I'm fine to change the file name to `Validator` if you have some concern.

Comment (PHILO-HE) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala]: The latest change is OK to me. Thanks!

**Comment by PHILO-HE** [comment]
@exmy, could you please take a review? There is some code change for CH. Thanks!

**Comment by exmy** [comment]
LGTM

**Comment by zhztheplayer** [comment]
LGTM. Thanks!

**Review by PHILO-HE — APPROVED** [review]
Thanks for your work!

**Pull request merged**